### PR TITLE
[8.1] Refactor TaskQueueDB for diracx

### PR DESCRIPTION
**Pull request by simon-mazenoux**
BEGINRELEASENOTES

ENDRELEASENOTES

Again, small refactor to avoid duplicating login between DIRAC and diracx

### CODE REVIEW

File: src/DIRAC/Core/Utilities/Graphs/QualityMapGraph.py [hunk 1]
```diff
@@ -171,7 +171,7 @@ def draw(self):
         cb = ColorbarBase(
             cax, cmap=self.cmap, norm=self.norms, boundaries=self.cbBoundaries, values=self.cbValues, ticks=self.cbTicks
         )
-        cb.draw_all()
+        cb._draw_all()
```
Comment (chrisburr) [inline src/DIRAC/Core/Utilities/Graphs/QualityMapGraph.py]: Shouldn't this have been fixed by https://github.com/DIRACGrid/DIRAC/pull/7210

Comment (simon-mazenoux) [reply src/DIRAC/Core/Utilities/Graphs/QualityMapGraph.py]: Sorry, I didn't check for such a pull request. Let me rebase and see if I still get the CI error.

File: src/DIRAC/WorkloadManagementSystem/DB/TaskQueueDB.py [hunk 2]
```diff
@@ -1232,3 +1188,55 @@ def getGroupShares():
         for group in groups:
             shares[group] = gConfig.getValue(f"/Registry/Groups/{group}/JobShare", DEFAULT_GROUP_SHARE)
         return shares
+
+
+def calculate_priority(tqDict, allTQsData, share, allowBgTQs):
```
Comment (fstagni) [inline src/DIRAC/WorkloadManagementSystem/DB/TaskQueueDB.py]: Now it's very easy to add a unit test. Would you provide one?

Comment (simon-mazenoux) [reply src/DIRAC/WorkloadManagementSystem/DB/TaskQueueDB.py]: Done, I also refactored the code to use list comprehensions and things like this. Given that I tested the code with both the old and new version, all the changes I've made should be fine.

**Review by chrisburr — APPROVED** [review]

**Pull request merged**